<?xml version="1.0" encoding="UTF-8"?>

<launch>
  <arg name="world"  default="$(find practica_servicios)/worlds/cafeteria.world"/>
  <!--deprecated argument, you should specify the parameters below-->
  <!-- They are described in tiago.urdf.xacro -->
  <arg name="robot"  default="steel"/>
  <arg name="arm"  default="$(eval {'iron': False}.get(arg('robot'), True))"/>
  <arg name="end_effector"  default="$(eval {'iron': 'false', 'steel': 'pal-gripper', 'titanium': 'pal-hey5'}.get(arg('robot'), 'pal-gripper'))"/>
  <arg name="ft_sensor"  default="$(eval {'titanium': 'schunk-ft'}.get(arg('robot'), 'false'))"/>
  <arg name="laser_model"  default="sick-571"/>
  <arg name="camera_model"  default="orbbec-astra"/>
  <arg name="multi" default=""/>
  <arg name="gzpose" default="-x 1.0 -y -4.0 -z 0.2 -R 0.0 -P 0.0 -Y 0.0"/>
  <arg name="gui" default="true"/>
  <arg name="debug" default="false"/>
  <arg name="public_sim" default="true"/>
  <arg name="tuck_arm" default="true"/>
  <arg name="recording" default="false"/>
  <arg name="extra_gazebo_args" default=""/>
  <arg name="use_moveit_camera" default="true"/>
  <arg name="use_dynamic_footprint" default="false"/>
  <arg name="pause" default="false" />
  <env name="LC_NUMERIC" value="C" />

  <!-- start up world -->
  <!-- <include file="$(find pal_gazebo_worlds)/launch/pal_gazebo.launch">
    <arg name="world" value="$(arg world)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="recording" value="$(arg recording)"/>
    <arg name="extra_gz_model_path" value="$(find tiago_gazebo)/models"/>
    <arg name="extra_gz_resource_path" value="$(find tiago_gazebo)"/>
    <arg name="extra_gazebo_args" value="$(arg extra_gazebo_args)"/>
  </include> -->

    <!-- include gazebo_ros launcher -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
            <arg name="world_name" value="$(arg world)" />
            <arg name="debug" value="$(arg debug)" />
            <arg name="gui" value="$(arg gui)" />
            <arg name="paused" value="$(arg pause)" />
            <arg name="use_sim_time" value="true" />
    </include>

  <!-- spawn robot in simulation -->
  <include file="$(find tiago_gazebo)/launch/tiago_spawn.launch">
    <arg name="arm" value="$(arg arm)"/>
    <arg name="end_effector" value="$(arg end_effector)"/>
    <arg name="ft_sensor" value="$(arg ft_sensor)"/>
    <arg name="laser_model" value="$(arg laser_model)"/>
    <arg name="camera_model" value="$(arg camera_model)"/>
    <arg name="multi" value="$(arg multi)"/>
    <arg name="gzpose" value="$(arg gzpose)"/>
  </include>

  <!-- bring up robot -->
  <include file="$(find tiago_gazebo)/launch/simulation_tiago_bringup.launch">
    <arg name="arm" value="$(arg arm)"/>
    <arg name="end_effector" value="$(arg end_effector)"/>
    <arg name="ft_sensor" value="$(arg ft_sensor)"/>
    <arg name="laser_model" value="$(arg laser_model)"/>
    <arg name="camera_model" value="$(arg camera_model)"/>
    <arg name="public_sim"  value="$(arg public_sim)"/>
    <arg name="use_moveit_camera" value="$(arg use_moveit_camera)"/>
    <arg name="use_dynamic_footprint" value="$(arg use_dynamic_footprint)"/>
  </include>

  <!-- point cloud throttle and filter -->
  <group unless="$(arg public_sim)">
    <include file="$(find pal_pcl_points_throttle_and_filter)/launch/throttle_filtering_points.launch">
      <arg name="cloud"  value="/xtion/depth_registered/points"/>
    </include>
  </group>

  <group if="$(arg tuck_arm)">
      <node name="tuck_arm" pkg="tiago_gazebo" type="tuck_arm.py" output="screen"/>
  </group>

  <node name="image_proc" pkg="image_proc" type="image_proc" ns="xtion/rgb"/>

  <node name="move_arm_server" pkg="practica_servicios" type="move_arm_server" output="screen" />

  <node name="cloud_processor" pkg="practica_servicios" type="geograsp_server" output="screen" />

  <node name="foo_throttler" type="throttle" pkg="topic_tools" args="messages /xtion/depth_registered/points 2  /throttle_filtering_points/filtered_points" />

  <rosparam command="load" file="$(find practica_servicios)/config/tiago_octomap_motions.yaml" />

</launch>
